### PR TITLE
Spec v2

### PR DESCRIPTION
**Pull request by notlee**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,10 +1,40 @@
...
+  "license": "MIT",
+  "author": "",
+  "type": "module",
+  "exports": {
+    ".": {
+      "browser": "./main.js"
     }
+  },
+  "browser": "./main.js",
+  "types": "./types/main.d.ts",
```
Comment (JakeChampion) [inline package.json:23]: This branch has more changes than the minimum required for origami v2 components such as this "types" property and the "exports" property

File: origami.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
     "description": "Tools to \"brand\" other Origami components.",
     "origamiType": "module",
-    "origamiVersion": 1,
+    "origamiVersion": 2,
```
Comment (JakeChampion) [inline origami.json:4]:
```suggestion
    "origamiVersion": "2.0",
```

Comment (JakeChampion) [reply origami.json:4]: I missed this in my previous review, apologies for that.
The version is meant to be a string which is a "2.0"

File: package.json [hunk 3]
```diff
@@ -1,10 +1,31 @@
...
+  "devDependencies": {
+    "eslint": "^7.9.0",
+    "eslint-config-origami-component": "^2.1.0",
+    "origami-ci-tools": "3.0.0-beta.6",
+    "remark": "^12.0.1",
+    "remark-lint": "^7.0.1",
+    "remark-preset-lint-origami-component": "^2.0.0-beta.1",
+    "stylelint": "^13.7.0",
+    "stylelint-config-origami-component": "^1.0.4",
+    "typescript": "^4.0.5"
```
Comment (chee) [inline package.json:26]: :eye: wots all this then

Comment (JakeChampion) [reply package.json:26]: This was from a prototype I did to see if we could use typescript to build .d.ts files from our JSDoc comments and if those .d.ts files would then be used automatically by code-editors

It worked in vscode, but, I never tried other code-editors

File: .github/workflows/npm-audit-fix.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  run:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-node@v2.1.4
+      with:
+        node-version: '12.x'
+        registry-url: 'https://registry.npmjs.org'
+    - run: npm install -g "npm@^7"
+    - uses: ybiquitous/npm-audit-fix-action@v2.1.5
```
Comment (JakeChampion) [inline .github/workflows/npm-audit-fix.yml:22]: The v3 of this action supports npm 7 whereas the v2 version does not
```suggestion
    - uses: ybiquitous/npm-audit-fix-action@v3
```

File: package.json (hunk 3 above)
Comment (JakeChampion) [reply package.json]: I've removed typescript dependency as it is not being used. We can have a discussion in the future about whether we would like to publish our jsdoc as typescript types

File: .github/workflows/test-origami-component.yml [hunk 5]
```diff
@@ -5,22 +5,11 @@ jobs:
...
-      with:
-        github-token: ${{ secrets.GITHUB_TOKEN }}
-    - run: npx origami-ci branch
-      env:
-        ORIGAMI_CI_NAME: ${{ steps.repo_name.outputs.REPO_NAME }}
-        ORIGAMI_CI_VERSION: 0.0.0
-        ORIGAMI_PULL_REQUEST: ${{ steps.findPr.outputs.pr }}
-        GITHUB_TOKEN: ${{ secrets.ORIGAMI_FOX_ACCESS_TOKEN }}
+    - run: npm install -g npm@^7
+    - run: npm install -g origami-build-tools@prerelease
```
Comment (JakeChampion) [inline .github/workflows/test-origami-component.yml:12]: Note: We should see if there is any work left before releasing obt v11

File: MIGRATION.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
+## Migration
+
+### Migrating from v3 to v4
+
+V4 has been dropped support for use through Bower. When using Sass, you will need to add `node_modules` to your `includePath` as well as either:
```
Comment (JakeChampion) [inline MIGRATION.md:5]:
```suggestion
V4 has dropped support for use through Bower. When using Sass, you will need to add `node_modules` to your `includePath` as well as either:
```

File: MIGRATION.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+- Update your code to use the fully qualified package name, i.e.: `@import "@financial-times/o-brand/main"`, or
+- Also `node_modules/@financial-times` to the Sass include path
```
Comment (JakeChampion) [inline MIGRATION.md:7]: Nice idea to list both these options 👍

File: origami.json [hunk 8]
```diff
@@ -1,9 +1,6 @@
 {
-    "description": "Tools to \"brand\" other Origami components.",
     "origamiType": "module",
```
Comment (JakeChampion) [inline origami.json:2]:
```suggestion
    "origamiType": "component",
```

File: .github/workflows/test-origami-component.yml [hunk 9]
```diff
@@ -5,22 +5,11 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
-    # Required for npx origami-ci branch, specifically for bundle-size-cli
-    - run: git fetch --prune --unshallow
     - uses: actions/setup-node@v2.1.5
       with:
         node-version: 12.x
```
Comment (JakeChampion) [inline .github/workflows/test-origami-component.yml:10]:
```suggestion
        node-version: 14.x
```

File: .github/workflows/npm-audit-fix.yml [hunk 10]
```diff
@@ -0,0 +1,27 @@
+name: npm audit fix
+
+on:
+  schedule:
```
Comment (notlee) [inline .github/workflows/npm-audit-fix.yml:4]: I think we decided to remove this workflow from projects, due to the number of PRs?

File: .github/workflows/test-origami-component.yml (hunk 5 above)
Comment (notlee) [reply .github/workflows/test-origami-component.yml:12]: Should we update this to a specific beta until then?

File: MIGRATION.md (hunk 7 above)
Comment (notlee) [reply MIGRATION.md:7]: Plus one this is nice! I think there's a missing word in the last line though? "Also [add]" 
```suggestion
- Update your code to use the fully qualified package name, i.e.: `@import "@financial-times/o-brand/main"`, or
- Also add `node_modules/@financial-times` to the Sass include path
```

File: MIGRATION.md [hunk 11]
```diff
@@ -0,0 +1,21 @@
+## Migration
+
+### Migrating from v3 to v4
+
+V4 has dropped support for use through Bower. When using Sass, you will need to add `node_modules` to your `includePath` as well as either:
+- Update your code to use the fully qualified package name, i.e.: `@import "@financial-times/o-brand/main"`, or
+- Also `node_modules/@financial-times` to the Sass include path
+
+### Migrating from v2 to v3
```
Comment (notlee) [inline MIGRATION.md:9]: Were these guides missing? Nice one adding them 👍

File: .github/workflows/npm-audit-fix.yml (hunk 10 above)
Comment (JakeChampion) [reply .github/workflows/npm-audit-fix.yml:4]: That's right, I'll remove it now 👍

File: MIGRATION.md (hunk 11 above)
Comment (chee) [reply MIGRATION.md:9]: i think you did

File: .github/workflows/test-origami-component.yml [hunk 12]
```diff
@@ -5,22 +5,11 @@ jobs:
...
-        github-token: ${{ secrets.GITHUB_TOKEN }}
-    - run: npx origami-ci branch
-      env:
-        ORIGAMI_CI_NAME: ${{ steps.repo_name.outputs.REPO_NAME }}
-        ORIGAMI_CI_VERSION: 0.0.0
-        ORIGAMI_PULL_REQUEST: ${{ steps.findPr.outputs.pr }}
-        GITHUB_TOKEN: ${{ secrets.ORIGAMI_FOX_ACCESS_TOKEN }}
+        node-version: 14.x
+    - run: npm install -g npm@^7
+    - run: npm install -g origami-build-tools@prerelease
```
Comment (chee) [inline .github/workflows/test-origami-component.yml:12]:
```suggestion
    - run: npm install -g origami-build-tools@11
```

File: MIGRATION.md (hunk 11 above)
Comment (notlee) [reply MIGRATION.md:9]: oh

File: .github/workflows/test-origami-component.yml (hunk 5 above)
Comment (JakeChampion) [reply .github/workflows/test-origami-component.yml]: We released v11 and updated the workflow to use v11 👍

**Review by JakeChampion — APPROVED** [review]

**Comment by origamiserviceuser** [comment]
:tada: This PR is included in version v4.0.0 :tada:

**Pull request merged**